iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake

```
@@ -111,3 +110,283 @@ Macro(install_package_files_and_export)
+            "${IOX_RPATH_PREFIX}/../../posh"
+            "${IOX_RPATH_PREFIX}/../../hoofs/platform"
+            "${IOX_RPATH_PREFIX}/../../binding_c"
+            # TODO: iox-#1287 iox-roudi is stored directly in build, despite it should be stored in iceoryx_posh, adjust paths so that this works too
+            "${IOX_RPATH_PREFIX}/hoofs"
+            "${IOX_RPATH_PREFIX}/posh"
+            "${IOX_RPATH_PREFIX}/hoofs/platform"
+            "${IOX_RPATH_PREFIX}/binding_c"
+            # TODO: END iox-#1287
+    )
+endMacro()
+
+Macro(iox_set_file_language)
+    set(switches USE_C_LANGUAGE)
+    set(multiArguments FILES)
+    cmake_parse_arguments(IOX "${switches}" "" "${multiArguments}" ${ARGN} )
+
+    if ( IOX_USE_C_LANGUAGE )
+        set_source_files_properties(${IOX_FILES} PROPERTIES LANGUAGE C)
+        set_target_properties(
+            ${IOX_TARGET}
+          PROPERTIES
+            C_STANDARD_REQUIRED ON
+            C_STANDARD 11
+        )
+    else()
+        set_source_files_properties(${IOX_FILES} PROPERTIES LANGUAGE CXX)
+        set_target_properties(
+            ${IOX_TARGET}
+          PROPERTIES
+            CXX_STANDARD_REQUIRED ON
+            CXX_STANDARD ${ICEORYX_CXX_STANDARD}
+        )
+    endif()
+endMacro()
+
+Macro(iox_add_executable)
+    set(switches USE_C_LANGUAGE PLACE_IN_BUILD_ROOT)
+    set(arguments TARGET STACK_SIZE)
+    set(multiArguments FILES LIBS INCLUDE_DIRECTORIES LIBS_QNX LIBS_LINUX LIBS_UNIX LIBS_WIN32 LIBS_APPLE 
+        BUILD_INTERFACE INSTALL_INTERFACE)
+    cmake_parse_arguments(IOX "${switches}" "${arguments}" "${multiArguments}" ${ARGN} )
+
+    add_executable(${IOX_TARGET} ${IOX_FILES})
+    target_include_directories(${IOX_TARGET} PRIVATE ${IOX_INCLUDE_DIRECTORIES})
+    target_link_libraries(${IOX_TARGET} ${IOX_LIBS})
+
+    if ( QNX )
+        target_link_libraries(${IOX_TARGET} ${IOX_LIBS_QNX})
+    elseif ( LINUX )
+        target_link_libraries(${IOX_TARGET} ${IOX_LIBS_LINUX})
+    elseif ( APPLE )
+        target_link_libraries(${IOX_TARGET} ${IOX_LIBS_APPLE})
+    elseif ( WIN32 )
+        target_link_libraries(${IOX_TARGET} ${IOX_LIBS_WIN32})
+    elseif ( UNIX )
+        target_link_libraries(${IOX_TARGET} ${IOX_LIBS_UNIX})
+    endif()
+
+    iox_set_rpath( IS_EXECUTABLE TARGET ${IOX_TARGET} )
+
+    if ( IOX_PLACE_IN_BUILD_ROOT )
+        set_target_properties(${IOX_TARGET} PROPERTIES
+            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
+        )
+    endif()
+
+    set(IOX_WARNINGS ${ICEORYX_WARNINGS})
+    if ( IOX_USE_C_LANGUAGE )
+        if("-Wno-noexcept-type" IN_LIST IOX_WARNINGS)
+            list(REMOVE_ITEM IOX_WARNINGS "-Wno-noexcept-type")
+        endif()
+    endif()
+
+    if ( IOX_USE_C_LANGUAGE )
+        iox_set_file_language( USE_C_LANGUAGE FILES ${IOX_FILES} )
+    else()
+        iox_set_file_language( FILES ${IOX_FILES} )
+    endif()
+
+    target_compile_options(${IOX_TARGET} ${IOX_WARNINGS} ${ICEORYX_SANITIZER})
```

Sanitizer Flags missing?
```suggestion
    target_compile_options(${IOX_TARGET} ${IOX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
```